fetch.js

```diff
@@ -195,44 +195,40 @@
       }
     }
 
-    if (support.blob) {
-      this.blob = function() {
-        var rejected = consumed(this)
-        if (rejected) {
-          return rejected
-        }
 
-        if (this._bodyBlob) {
-          return Promise.resolve(this._bodyBlob)
-        } else if (this._bodyFormData) {
-          throw new Error('could not read FormData body as blob')
-        } else {
-          return Promise.resolve(new Blob([this._bodyText]))
-        }
+    this.blob = function() {
+      var rejected = consumed(this)
+      if (rejected) {
+        return rejected
       }
 
-      this.arrayBuffer = function() {
-        return this.blob().then(readBlobAsArrayBuffer)
+      if (this._bodyBlob) {
+        return Promise.resolve(this._bodyBlob)
+      } else if (this._bodyFormData) {
+        throw new Error('could not read FormData body as blob')
+      } else {
+        return new Promise(function(resolve){
+          resolve(new Blob([this._bodyText]))
```

Comment:
I was happier with the previous approach. I expected the `blob()` method to throw `Blob is not defined`, not reject.